lib/ansible/plugins/lookup/ssm.py

```diff
@@ -0,0 +1,83 @@
+# (c) 2016, Bill Wang <ozbillwang(at)gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils.ec2 import HAS_BOTO3
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+try:
+    from botocore.exceptions import ClientError
+    import boto3
+except ImportError:
+    pass  # will be captured by imported HAS_BOTO3
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+
+        ret = {}
+        response = {}
+
+        ssm_args = terms[0].split()
```

Comment:
Just use `variables` to get region

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/__init__.py#L76-L85

to get `lookup('ssm', key, region='us-east-1')` or whatever

Reply:
What if we don't have an environment variable set, but still want to pass a region? By default, boto3 can pick up a region from  ~/.aws/config file, but users _will_ want to override it, if they need to lookup parameters from another region! (Each AWS region has it's own set of SSM Parameters)